### PR TITLE
CMHierarchySetting: Resource to configure Site Hierarchy Settings

### LINKED ISSUE #108: CMHierarchySettings: New resource proposal
### Description
New resource that sets all parameters of the Set-CMHierarchySetting cmdlet

### Proposed properties
All parameters of the Set-CMHierarchySetting cmdlet

### Special considerations or limitations
Get-CMHierarchySetting does not return some set values. Additional CIM queries are required.

### PR DESCRIPTION
**Pull request by nyanhp**
#### Pull Request (PR) description

Enabling configuration of every setting that `Set-CMHierarchySetting` exposes.

#### This Pull Request (PR) fixes the following issues

- Fixes #108

#### Task list

- [x] Added an entry to the change log under the Unreleased section of the
      file CHANGELOG.md. Entry should say what was changed and how that
      affects users (if applicable), and reference the issue being resolved
      (if applicable).
- [x] Resource documentation added/updated in README.md.
- [x] Resource parameter descriptions added/updated in README.md, schema.mof
      and comment-based help.
- [x] Comment-based help added/updated.
- [x] Localization strings added/updated in all localization files as appropriate.
- [x] Examples appropriately added/updated.
- [x] Unit tests added/updated. See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [ ] Integration tests added/updated (where possible). See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [x] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

- - -
This change is [Reviewable](https://reviewable.io/reviews/dsccommunity/configmgrcbdsc/109)

### CODE REVIEW

**Comment by nyanhp** [comment]
It is unclear to me why the build fails in this repo. A local build with `.\build.ps1 -Resolve -CodeCoverageThreshold 0` works fine. I was able to import and test the module in my MECM lab. Let me know what I need to correct to resolve this issue.

File: source/DSCResources/DSC_CMHierarchySetting/en-US/DSC_CMHierarchySetting.strings.psd1 [hunk 1]
```diff
@@ -0,0 +1,7 @@
+ConvertFrom-StringData @'
+RetrieveSettingValue = Getting current hierarchy settings.
```
Comment (NEllis280) [inline source/DSCResources/DSC_CMHierarchySetting/en-US/DSC_CMHierarchySetting.strings.psd1:2]: spacing before = throughout

File: CHANGELOG.md [hunk 2]
```diff
@@ -39,6 +39,7 @@ version of DSCResource.Common
 - Added CMClientSettingsStateMessaging resource
 - Added CMClientSettingsUserDeviceAffinity resource
 - Added CMSiteConfiguration resource
+- Added CMHierarchySetting resource
```
Comment (NEllis280) [inline CHANGELOG.md:42]: Please move to the unreleased section

File: README.md [hunk 3]
```diff
@@ -2176,6 +2177,32 @@ you are using apply and auto correct.
 - [CMSiteConfiguration_Cas](Source\Examples\Resources\CMSiteConfiguration\CMSiteConfiguration_Cas.ps1)
 - [CMSiteConfiguration_Primary](Source\Examples\Resources\CMSiteConfiguration\CMSiteConfiguration_Primary.ps1)
 
+### CMHierarchySetting
```
Comment (NEllis280) [inline README.md:2180]: Please add the new resource to supported resources under the reverse dsc section

File: source/DSCResources/DSC_CMHierarchySetting/en-US/DSC_CMHierarchySetting.strings.psd1 (hunk 1 above)
Comment (nyanhp) [reply source/DSCResources/DSC_CMHierarchySetting/en-US/DSC_CMHierarchySetting.strings.psd1:2]: Done

File: source/DSCResources/DSC_CMHierarchySetting/DSC_CMHierarchySetting.psm1 [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+                    -f $property.Key, $property.Value, $state[$property.key])
+            $result = $false
+        }
+    }
+
+    Write-Verbose -Message ($script:localizedData.TestState -f $result)
+    return $result
+}
+
+Export-ModuleMember -Function *-TargetResource
```
[the remark this answers is not in the document]
Comment (nyanhp) [reply source/DSCResources/DSC_CMHierarchySetting/DSC_CMHierarchySetting.psm1:357]: Done

File: CHANGELOG.md (hunk 2 above)
Comment (nyanhp) [reply CHANGELOG.md]: Done

File: README.md (hunk 3 above)
Comment (nyanhp) [reply README.md:2180]: Done

**Review by NEllis280 — APPROVED** [review]
Approved, pending resolution of merge conflicts.

File: tests/Unit/ConfigMgrCBDsc.ReverseDsc.tests.ps1 [hunk 5]
```diff
@@ -6352,6 +6495,7 @@ InModuleScope $script:subModuleName {
                 Mock -CommandName Invoke-DscResource -MockWith { $invokeCMClientSettingsStateMessaging } -ParameterFilter { $Name -eq 'CMClientSettingsStateMessaging' }
                 Mock -CommandName Invoke-DscResource -MockWith { $invokeCMClientSettingsUserDeviceAffinity } -ParameterFilter { $Name -eq 'CMClientSettingsUserDeviceAffinity' }
                 Mock -CommandName Invoke-DscResource -MockWith { $invokeSiteConfigurationCas } -ParameterFilter { $Name -eq 'CMSiteConfiguration' }
+                Mock -CommandName Invoke-DscResource -MockWith { $invokeHierarchySetting } -ParameterFilter { $Name -eq 'CMHierarchySetting' }
```
Comment (NEllis280) [inline tests/Unit/ConfigMgrCBDsc.ReverseDsc.tests.ps1:6499]: On line 6503 below, need to change 48 to 49 (now that the other branch has been merged in.) in order to resolve failing pester tests.

**Pull request merged**